instrumentation-tests/src/androidTest/java/com/mapbox/navigation/instrumentation_tests/utils/location/LocationUpdate.kt

```diff
@@ -0,0 +1,30 @@
+package com.mapbox.navigation.instrumentation_tests.utils.location
+
+import com.mapbox.navigation.testing.ui.BaseCoreNoCleanUpTest
+import kotlinx.coroutines.coroutineScope
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.launch
+
+suspend fun BaseCoreNoCleanUpTest.stayOnPosition(
+    latitude: Double,
+    longitude: Double,
+    frequencyHz: Int = 10,
+    block: suspend () -> Unit
+) {
+    coroutineScope {
+        val updateLocations = launch {
+            while (true) {
+                mockLocationUpdatesRule.pushLocationUpdate {
```

Comment:
it won't happen with this implementation because location update works only for the given scope:
```
stayOnPosition(position1) {
   // position 1 is emulated here
}
// no position is emulated here
stayOnPosition(position2) {
  // position 2 is emulated here
}
```